woocommerce/woocommerce
# DateTimePickerControl updates

### All Submissions:






### Changes proposed in this Pull Request:



This PR replaces the underlying implementation of `DateTimePickerControl` to be based on `__experimentalInputControl` from `@wordpress/components` instead of `DateInput` from `@woocommerce/components`'s `Calendar`. There really wasn't anything we were gaining from using `DateInput`, and that sub-component was never really designed to be reused outside of `Calendar`. The styling was off, and it didn't support the props we need for full integration in the product form. The product form is using `__experimentalInputControl` for other fields, so by using that under the hood in this component we will gain additional consistency in behavior and look-and-feel.

In addition to changing the underlying implementation, the following props were added:
- `label`
- `placeholder`
- `help`

Note: the following issues exist and will be addressed in follow-up PRs:
- The positioning of the popup picker. It should be left-aligned. However, there appears to be a bug in the underlying `Dropdown` component in the version we are using (see https://github.com/WordPress/gutenberg/pull/41361, https://github.com/WordPress/gutenberg/pull/43377). Upgrading `@wordpress/components` may address this.
- There may be minor styles tweaks needed once this component is integrated in the product form (see https://github.com/woocommerce/woocommerce/pull/34538)
- The use of `moment` makes things easy, since we are used to using `moment`. However, `moment` has performance issues (seen when running the unit tests in CI) and isn't really the way forward. We should consider replacing it with a smaller, more performant library (from `moment`'s website: https://momentjs.com/docs/#/-project-status/)
    - The poor performance of `moment` is the major reason why two of the tests have their timeout increased. The tests run extremely slow in our CI environment and timeout otherwise.
- Two of the unit tests are skipped because they do not work with Jest's fake timers. I believe it we upgrade [Jest](https://jestjs.io/), [React Testing Library](https://testing-library.com/docs/react-testing-library/intro/), [Testing Library User Event](https://testing-library.com/docs/user-event/intro/), we will be able to re-enable these tests since there is additional support to advance the fake timers from the tests.

### How to test the changes in this Pull Request:

1. Run unit tests: `pnpm --filter=@woocommerce/components run test -- src/date-time-picker-control`
2. Interact with Storybook stories and make sure component functions and looks reasonable: `pnpm --filter=@woocommerce/storybook storybook`

### Other information:




### FOR PR REVIEWER ONLY:
